veniq/dataset_collection/augmentation.py

```diff
@@ -136,55 +122,125 @@ def _is_match_to_the_conditions(
         return False
 
 
-def determine_type(method_node: ASTNode):
+def check_whether_method_has_return_type(
+        method_decl: AST,
+        var_decls: typing.Set[str]) -> InlineTypesAlgorithms:
     """
-
-    :param method_node: Method declaration
-    :return: int - type
+    Run function to check whether Method declaration can be inlined
+    :param method_decl: method, where invocation occurred
+    :param var_decls: set of variables for found invoked method
+    :return: enum InlineTypesAlgorithms
     """
+    names = get_variables_decl_in_node(method_decl)
 
-    if not method_node.parameters:
-        return 0
-    else:
-        return -1
+    var_decls_original = set(names)
+    intersected_names = var_decls & var_decls_original
+    # if we do not have intersected name in target method and inlined method
+    # and if we do not have var declarations at all
+    if not var_decls or not intersected_names:
+        return InlineTypesAlgorithms.WITHOUT_RETURN_WITHOUT_ARGUMENTS
+
+    return InlineTypesAlgorithms.DO_NOTHING
+
+
+def get_variables_decl_in_node(
+        method_decl: AST) -> List[str]:
+    names = []
+    for x in method_decl.get_proxy_nodes(ASTNodeType.VARIABLE_DECLARATOR):
```

Comment:
`VARIABLE_DECLARATION` and `LOCAL_VARIABLE_DECLARATION` does not have `name` field. They have array `declarators`. Each `VARIABLE_DECLARATOR` has `name` field. `VARIABLE_DECLARATION` and `LOCAL_VARIABLE_DECLARATION` has field `names` to get array of `name` of each `declorator`.

I think now you can avoid `hasattr` function.